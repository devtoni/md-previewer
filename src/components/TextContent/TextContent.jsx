import React from 'react'

import styles from './TextContent.scss'

const TextContent = ({onHandleChange}) => {
  const handleUpdate = (e) => {
    onHandleChange(e.target.value)
  }
  return (
    <textarea
      className={styles.Textarea}
      onChange={handleUpdate}
      defaultValue='Write here' />
  )
}

export default TextContent
