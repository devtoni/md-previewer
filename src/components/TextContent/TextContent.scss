.Textarea {
  width: 90%;
  max-width: 600px;
  min-height: 500px;
  border: 3px solid #cccccc;
  padding: 5px;
  resize: none;
}