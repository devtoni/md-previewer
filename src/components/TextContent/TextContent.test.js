import React from 'react'
import { shallow } from 'enzyme'

import TextContent from './TextContent'

test('should render correctly', () => {
  const component = shallow(<TextContent />)
  expect(component).toMatchSnapshot()
})
