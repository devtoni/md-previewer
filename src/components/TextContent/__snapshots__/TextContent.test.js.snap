// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <textarea
    cols="30"
    defaultValue="Write here"
    onChange={[Function]}
    rows="10"
  />,
  "nodes": Array [
    <textarea
      cols="30"
      defaultValue="Write here"
      onChange={[Function]}
      rows="10"
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <TextContent />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <textarea
          cols="30"
          defaultValue="Write here"
          onChange={[Function]}
          rows="10"
        />,
        "_debugID": 2,
        "_renderedOutput": <textarea
          cols="30"
          defaultValue="Write here"
          onChange={[Function]}
          rows="10"
        />,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TextContent />,
}
`;
